### PR TITLE
Update dependencies

### LINKED ISSUE #398: Update ops dependencies for v2023.3
This issue is for .3, not .2.

### PR DESCRIPTION
**Pull request by lognaturel**
Closes #398

I have standardized on pinning the major and minor versions for standard Docker images. This feels like the right balance between predictability and getting updates.

#### What has been done to verify that this works as intended?
It's on the dev server.

#### Why is this the best possible solution? Were any other approaches considered?
I also considered pinning the patch version or a SHA to ensure consistency. But i think this is a good balance.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

The only minor version changed is for node. I haven't looked at the release notes.

#### Does this change require updates to documentation? If so, please file an issue [here](https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:

- [x] verified that any code or assets from external sources are properly credited in comments or that everything is internally sourced

### CODE REVIEW

**Comment by matthew-white** [comment]
> The only minor version changed is for node. I haven't looked at the release notes.

If we wanted additional verification of this change, we could update Node in central-backend and central-frontend. These look to be the PRs from the last update: getodk/central-backend#786, getodk/central-frontend#746. The main benefit of that would be getting to run Backend and Frontend tests in CircleCI under the new version of Node. But also regression testing would probably catch most issues.

**Review by matthew-white — APPROVED** [review]

File: postgres14.dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM postgres:14
+FROM postgres:14.8
```
Comment (matthew-white) [inline postgres14.dockerfile:1]: I didn't realize this at first, but some users will have 14.7 or maybe even 14.6 installed, so this will be an upgrade for them. (I initially thought that this line was just pinning the current version.) I don't see why this change would be an issue, but I thought I'd make a note of it. Nothing special needs to happen when upgrading within a major version of Postgres, right?

Comment (matthew-white) [reply postgres14.dockerfile:1]: test.central.getodk.org is at Postgres 14.6, so we'll get a test of 14.6→14.8.

Comment (lognaturel) [reply postgres14.dockerfile:1]: > Nothing special needs to happen when upgrading within a major version of Postgres, right

It shouldn't be the case but mistakes happen so that's why the pinning seems like a good idea moving forward.
